### DISCUSSION TITLE
Help understanding re-renders when using `formState: {errors}`

### DISCUSSION THREAD

**Discussion by VoloDziu** [react-hook-form/react-hook-form#11235] [Q&A]
Hi folks,

I was wondering if somebody could help me understand why there's an extra re-render whenever I use `errors` from `formState`. 

From what I can tell from [this issue](https://github.com/react-hook-form/react-hook-form/issues/11167), it seems to be expected behaviour, not a bug.

I also saw in [this thread](https://github.com/orgs/react-hook-form/discussions/9987#discussioncomment-5175200) that this behaviour might be related to the `errors` being a Proxy, but not sure how that answer applies to a simple form like I have in my example.

[Example on codesandbox](https://codesandbox.io/p/sandbox/react-hook-form-get-started-ts-forked-37xjwh?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clph3e77m00073b6i9c3wsf46%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clph3e77m00033b6ids9tpc1t%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clph3e77m00043b6ismbdmom7%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clph3e77m00063b6i5xxkp7kk%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clph3e77m00033b6ids9tpc1t%2522%253A%257B%2522id%2522%253A%2522clph3e77m00033b6ids9tpc1t%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clph3k9m300033b6hxd6dlo0t%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A28%252C%2522startColumn%2522%253A40%252C%2522endLineNumber%2522%253A28%252C%2522endColumn%2522%253A40%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252Findex.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clph3k9m300033b6hxd6dlo0t%2522%257D%252C%2522clph3e77m00063b6i5xxkp7kk%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clph3e77m00053b6ir9lsrca4%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%2522%257D%255D%252C%2522id%2522%253A%2522clph3e77m00063b6i5xxkp7kk%2522%252C%2522activeTabId%2522%253A%2522clph3e77m00053b6ir9lsrca4%2522%257D%252C%2522clph3e77m00043b6ismbdmom7%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clph3e77m00043b6ismbdmom7%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

Just to illustrate what I mean by "re-renders":
| test case | with `errors`| without accessing `errors` (comment out) |
| --- | ---- | ---- |
| submitting invalid form | one re-render | zero re-renders | 
| submitting valid form | two re-renders | zero re-renders |

Thanks!

**Comment by BigSamu** [comment]
I am having the same issue. Not sure why.
